apple/OmnivoreKit/Sources/Services/AudioSession/AudioSession.swift

```diff
@@ -225,6 +326,12 @@ public class AudioSession: ObservableObject {
       throw BasicError.message(messageText: "audioFetch failed. no response or bad status code.")
     }
     print("httpResponse: ", httpResponse)
+    if let httpResponse = result?.1 as? HTTPURLResponse, httpResponse.statusCode == 202 {
```

Comment:
Do we need to poll the API here so we know when the download is complete?

Reply:
Oh yeah that is a good idea, maybe we could do like a permanent snackbar in that state or a different indicator than the spinner

Or maybe just some data on the miniplayer...